docs/how-to/rocm-for-ai/train-a-model.rst

```diff
@@ -2,139 +2,503 @@
+- Flash Attention (FA) 2
 
-#. Copy the model to every device so each device can process its local batches independently.
+- Fused kernels
 
-#. Run a forward pass, then a backward pass, and output the gradient of the weights with respect to the loss of the
-   model for that local batch. This happens in parallel on multiple devices.
+- Pre-training
 
-#. Synchronize the local gradients computed by each device and combine them to update the model weights. The updated
-   weights are then redistributed to each device.
+.. _amd-megatron-lm-model-support:
 
-In DDP training, each process or worker owns a replica of the model and processes a batch of data, then the reducer uses
-``allreduce`` to sum up gradients over different workers.
+The following models are pre-optimized for performance on the AMD Instinct MI300X accelerator.
 
-See the following developer blogs for more in-depth explanations and examples.
+* Llama 2 7B
 
-*  `Multi GPU training with DDP — PyTorch Tutorials <https://pytorch.org/tutorials/beginner/ddp_series_multigpu.html>`_
+* Llama 2 70B
 
-*  `Building a decoder transformer model on AMD GPUs — ROCm Blogs
-   <https://rocm.blogs.amd.com/artificial-intelligence/decoder-transformer/README.html#distributed-training-on-multiple-gpus>`_
+* Llama 3 8B
 
-.. _rocm-for-ai-pytorch-fsdp:
+* Llama 3 70B
 
-PyTorch FSDP
-------------
+* Llama 3.1 8B
 
-As noted in :ref:`PyTorch distributed <rocm-for-ai-pytorch-distributed>`, in DDP model weights and optimizer states
-are evenly replicated across all workers. Fully Sharded Data Parallel (FSDP) is a type of data parallelism that shards
-model parameters, optimizer states, and gradients across DDP ranks.
+* Llama 3.1 70B
 
-When training with FSDP, the GPU memory footprint is smaller than when training with DDP across all workers. This makes
-the training of some very large models feasible by allowing larger models or batch sizes to fit on-device. However, this
-comes with the cost of increased communication volume. The communication overhead is reduced by internal optimizations
-like overlapping communication and computation.
+Prerequisite system validation steps
+====================================
 
-For a high-level overview of how FSDP works, review `Getting started with Fully Sharded Data Parallel
-<https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html#how-fsdp-works>`_.
+Complete the following system validation and optimization steps to set up your system before starting training.
 
-For detailed training steps, refer to the `PyTorch FSDP examples
-<https://github.com/pytorch/examples/tree/main/distributed/FSDP>`_.
+Disable NUMA auto-balancing
+---------------------------
 
-.. _rocm-for-ai-deepspeed:
+Generally, application performance can benefit from disabling NUMA auto-balancing; however, there are
+certain types of workloads where doing so might be detrimental to performance.
```

Comment:
```suggestion
Generally, application performance can benefit from disabling NUMA auto-balancing. However, it might be detrimental to performance with certain workloads.
```